hicommonwealth/commonwealth
# Add community threads/addresses count

## Link to Issue
Closes: #6995

## Description of Changes
- Adds migration that adds `community.thread_count` and `community.address_count` columns and sets computed values
- Adds sequelize hooks that increment `community.thread_count` and `community.address_count` when threads/addresses are created or destroyed
- Adds logic to the `yarn recompute-counts` script for manual sync the counts

## Test Plan
- Run migrations
- Go to Etheruem
- Run query: `SELECT id, thread_count, address_count FROM "Communities" WHERE id = 'ethereum'` – note the number of threads
- Create a thread
- Run query– confirm that `thread_count` incremented by 1
- Delete thread
- Run query– confirm that `thread_count` decremented by 1
- Click profile –> Connect a new address
- Run query-> confirm that `address_count` incremented by 1

## Deployment Plan
- Nothing to do– migration will compute and set counts

## Other Considerations
- Just learned that `instance.destroy()` is the only way to ensure the `afterDestroy` sequelize hook is executed. `model.destroy({ where })` does not trigger the afterdestroy hook: https://github.com/sequelize/sequelize/issues/3628
